api/v1/openapi.yaml

```diff
@@ -0,0 +1,932 @@
+    required: true
+    in: path
+    type: integer
+  service-address:
+    name: address
+    description: Service address configuration
+    in: body
+    schema:
+      "$ref": "#/definitions/FrontendAddress"
+  service-config:
+    name: config
+    description: Service configuration
+    in: body
+    schema:
+      "$ref": "#/definitions/Service"
+  ipam-ip:
+    name: ip
+    description: IP address
+    in: path
+    required: true
+    type: string
+  ipam-family:
+    name: family
+    in: query
+    type: string
+    enum:
+    - ipv4
+    - ipv6
+definitions:
+  Endpoint:
+    description: Endpoint
+    type: object
+    properties:
+      id:
+        description: Unique identification
+        type: integer
+      docker-id:
+        description: Docker container ID
+        type: string
+      docker-endpoint-id:
+        description: Docker endpoint ID
+        type: string
+      docker-network-id:
+        description: Docker network ID
+        type: string
+      interface-name:
+        description: Name of network device
+        type: string
+      interface-index:
+        description: Index of network device
+        type: integer
+      state:
+        description: Current state of endpoint
+        type: string
+        enum:
+        - creating
+        - created
+        - joining
+        - ready
+      mac:
+        description: MAC address
+        type: string
+      host-mac:
+        description: MAC address
+        type: string
+      addressing:
+        "$ref": "#/definitions/EndpointAddressing"
+      identity:
+        description: Security identity
+        "$ref": "#/definitions/Identity"
+  EndpointResponse:
+    description: Response to an endpoint PUT request
+    type: object
+    properties:
+      ipam-config:
+        description: IPAM configuration to be associated with the new endpoint
+        "$ref": "#/definitions/IPAM"
+  IPAM:
+    description: IPAM configuration of an endpoint
+    type: object
+    required:
+      - endpoint
+      - host-addressing
+    properties:
+      endpoint:
+        "$ref": "#/definitions/EndpointAddressing"
+      host-addressing:
+        "$ref": "#/definitions/NodeAddressing"
+  EndpointAddressing:
+    description: Addressing information of an endpoint
+    type: object
+    properties:
+      ipv4:
+        description: IPv4 address
+        type: string
+      ipv6:
+        description: IPv6 address
+        type: string
+  Address:
+    description: IP address
+    type: string
+  Identity:
+    description: Security identity
+    type: object
+    properties:
+      id:
+        description: Unique identifier
+        type: integer
+      labels:
+        description: Labels describing the identity
+        "$ref": "#/definitions/Labels"
+  Labels:
+    description: Set of labels
+    type: array
+    items:
+      type: string
+  LabelConfiguration:
+    description: Label configuration of an endpoint
+    type: object
+    properties:
+      all:
```

Comment:
description: All labels, used to distinguish which probes labels can be disabled/deleted by the user without them to reappearing next time a probe pools the same labels.